[v3,linux] fix: enable to run natively in wayland

stendler opened the pull request:
<!--
READ CAREFULLY: Before submitting your PR, please ensure you have created an issue for your PR.
It is essential that you do this so that we can discuss the proposed changes before you spend time on them.
If you do not create an issue, your PR may be rejected without review.
If a relevant issue already exists, please reference it in your PR by including `Fixes #<issue number>` in your PR description.
-->

# Description

Incorporated from this v2 change:
https://github.com/wailsapp/wails/pull/1811

## Type of change
  
- [x] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?
  
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration using `wails doctor`.

- [x] Linux
  
Verified using `xeyes`: if the eyes follow the mouse when entering the wails app window, then it is running on X11; if they stop following when entering the window it is running on wayland.

## Test Configuration

```
# Build Environment
┌─────────────────────────────────────────────────────────┐
| Wails CLI    | v3.0.0-alpha.4                           |
| Go Version   | go1.22.4                                 |
| Revision     | 5fcebd02741eea738dfcb7fc43d8d86f139293fe |
| Modified     | true                                     |
| -buildmode   | exe                                      |
| -compiler    | gc                                       |
| CGO_CFLAGS   |                                          |
| CGO_CPPFLAGS |                                          |
| CGO_CXXFLAGS |                                          |
| CGO_ENABLED  | 1                                        |
| CGO_LDFLAGS  |                                          |
| GOAMD64      | v1                                       |
| GOARCH       | amd64                                    |
| GOOS         | linux                                    |
| vcs          | git                                      |
| vcs.modified | true                                     |
| vcs.revision | 5fcebd02741eea738dfcb7fc43d8d86f139293fe |
| vcs.time     | 2024-07-02T08:49:19Z                     |
└─────────────────────────────────────────────────────────┘

# Dependencies
┌────────────────────────────────────────┐
| npm        | 10.8.2                    |
| pkg-config | 2.1.1-1                   |
| webkit2gtk | 2.44.2-1                  |
| gcc        | 14.1.1+r58+gfc9fb69ad62-1 |
| gtk3       | 1:3.24.43-1               |
└─────── * - Optional Dependency ────────┘
```

# Checklist:

- [ ] I have updated `website/src/pages/changelog.mdx` with details of this PR
- [ ] My code follows the general coding style of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->

## Summary by CodeRabbit

- **New Features**
	- Updated changelog to include a fix for running natively in Wayland on Linux, enhancing cross-platform compatibility for Linux users.
  
- **Bug Fixes**
	- Improved initialization logic for Linux applications to prevent conflicts with existing environment variable values.

<!-- end of auto-generated comment: release notes by coderabbit.ai -->

atterpac commented:
This looks fine to me. Have you noticed any other artifacts of using wayland backend in the webkit window?

stendler commented:
Haven't noticed any issues with wayland so far. Though, I've only used the cgo version. The purego code seems to set `_ = os.Setenv("GDK_BACKEND", "x11")` without any checks as well. But that also does not seem on par feature wise?

leaanthony commented:
Thanks 🙏 Please could you add an entry to the changelog located at `mkdocs-website/docs/changelog.md`? Thanks!

stendler commented:
Done!

leaanthony commented:
Can you please fix the merge conflicts 👍

stendler commented:
I've rebased and moved the changelog line to the new unreleased section.

Although now I seem to have trouble building on linux (likely due to changes in https://github.com/wailsapp/wails/pull/3667):

```sh
 WARNING  -: # github.com/wailsapp/wails/v3/pkg/application                                                                                                                                                                                                                                                         
          wails/v3/pkg/application/webview_window_linux.go:383:69: could not determine kind of name for C.GDK_ALL_EVENTS_MASK
          wails/v3/pkg/application/webview_window_linux.go:381:69: could not determine kind of name for C.GDK_ENTER_NOTIFY_MASK
          wails/v3/pkg/application/webview_window_linux.go:381:93: could not determine kind of name for C.GDK_LEAVE_NOTIFY_MASK
          wails/v3/pkg/application/webview_window_linux.go:381:29: could not determine kind of name for C.GtkWidget
          wails/v3/pkg/application/webview_window_linux.go:381:3: could not determine kind of name for C.gtk_widget_set_events
 WARNING  /myapp/wails/v3/pkg/application/application_linux.go:15:8: could not import C (no metadata for C)                                                                                                                                                             
 WARNING  /myapp/wails/v3/pkg/application/webview_window_linux.go:285:25: undefined: options                                                                                                                                                                            
 WARNING  /myapp/wails/v3/pkg/application/webview_window_linux.go:381:42: undefined: unsafe                                                                                                                                                                             
 WARNING  /myapp/wails/v3/pkg/application/webview_window_linux.go:383:42: undefined: unsafe  
```
